src/twisted/web/http.py

```diff
@@ -2250,6 +2250,9 @@ def lineReceived(self, line):
         Called for each line from request until the end of headers when
         it enters binary mode.
         """
+        assert (
+            not self._handlingRequest
+        ), "when handling a request, we MUST be in raw mode to buffer the incoming data without parsing it"
```

Comment:
Maybe something like this... to check these type of errors in the wild.

I might have hit this bug in the wild, without knowing why.

I think that some .NET http client implementation have some HTTP pipelining and other IIS optimization and will send more requests in the same TCP package.


```suggestion
        # This is here since September 2023 to make sure we don't accidentally handle requests out of order.
        # If no bugs are reported in the next two year,
        # this line can be removed.
        assert (
            not self._handlingRequest
        ), "when handling a request, we MUST be in raw mode to buffer the incoming data without parsing it"
```

Reply:
This code should be impossible to hit in the wild and we have tests to verify that this is the case, I don't think random asserts like this are helpful.